core/jobs.py

```diff
@@ -192,8 +190,18 @@ def register_completion(cls, job_id, output_list):
 
         model.status_code = STATUS_CODE_COMPLETED
         model.time_finished_msec = utils.get_current_time_in_millisecs()
+        model.output = cls._compress_outputs(output_list)
+        model.put()
+
+        cls._post_completed_hook(job_id)
+
+    @classmethod
+    def _compress_outputs(cls, output_list, max_output_length_chars=900000):
```

Comment:
Actually, I don't think the max_output_length_chars should be configurable. It has to be a specific number because App Engine instances are limited to 1 MB.